packages/compiler/core/emitter-utils.ts

```diff
@@ -13,6 +15,13 @@ export interface EmitFileOptions {
  * @param options File Emitter options
  */
 export async function emitFile(program: Program, options: EmitFileOptions): Promise<void> {
+  // ensure path exists
+  const outputFolder = path.dirname(options.path);
+  try {
+    await mkdirp(outputFolder);
```

Comment:
This should go through the host. The catch is masking the issue. It should work with the virtual test file system. Perhaps we need to add mkdirp to the host interface.

Reply:
It's possible that it wouldn't fail and a test would start making real directories like this even though it's using a virtual file system. There shouldn't be any file IO done at all without going through the host interface.